src/steps/save_state.go

```diff
@@ -18,6 +20,10 @@ func saveState(runState *RunState) {
 		log.Fatal(err)
 	}
 	filename := getRunResultFilename(runState.Command)
+	err = os.MkdirAll(path.Dir(filename), 0644)
```

Comment:
Maybe add this logic to the `getRunResultFilename` method? It returns a filename that is guaranteed to be in a directory that exists, so it can be written to.